src/bson_util/async_encoding.rs

```diff
@@ -1,7 +1,7 @@
-use bson::Document;
 use futures::io::{AsyncRead, AsyncReadExt, AsyncWrite, AsyncWriteExt};
 
 use crate::{
+    bson::Document,
```

Comment:
As a sanity check to make sure the re-exporting of BSON worked, I changed all the imports in the driver to use BSON components from there rather than from the external package.